odoo/addons/base/controllers/avatar.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from hashlib import sha1
+
+from odoo.http import Controller, request, route, Response, local_redirect
+from odoo import tools
+
+# show avatar if no image or showAvatar 1
+# show image if image
+
+class Avatar(Controller):
+    @route(['/web/avatar/<int:user_id>'], auth='user', methods=['GET'])
+    def get_avatar(self, user_id=1):
+        if 'isPartner' in request.params and request.params['isPartner'] == '1':
+            user = request.env['res.users'].search([('partner_id', '=', user_id), ('groups_id.id', '=', request.env.ref('base.group_user').id)]).exists()
+        else:
+            user = request.env['res.users'].browse(user_id).exists()
+        if user:
+            if 'consume' in request.params and request.params['consume'] == '1' and user.image_128:
+                return local_redirect(f'/web/image?model=res.users&id={user.id}&field=image_128')
+            else:
+                avatar = _generate_avatar_svg(user.name, user.create_date)
+                return Response(avatar, mimetype='image/svg+xml')
+        return request.not_found()
+
+def _generate_avatar_svg(name, seed=False):
+    initials = tools.html_escape(_generate_initials(name))
+    bgcolor = tools.html_escape(_generate_bgcolor(f"{name}{seed.timestamp()}"))
+    if initials and bgcolor:
+        return (
+            "<?xml version='1.0' encoding='UTF-8' ?>"
+            "<svg height='180' width='180' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'>"
+            f"<rect fill='{bgcolor}' height='180' width='180'/>"
+            f"<text fill='#ffffff' font-size='64' text-anchor='middle' x='90' y='106' font-family='sans-serif'>{initials}</text>"
+            "</svg>"
+        )
+    else:
+        return _generate_default_avatar()
+
+def _generate_default_avatar(bgcolor='hsl(230, 0%, 90%)'):
+    return (
+        "<?xml version='1.0' encoding='UTF-8' ?>"
+        "<svg height='180' width='180' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'>"
+        f"<rect fill='{tools.html_escape(bgcolor)}' height='180' width='180'/>"
+        "<text fill='#ffffff' font-size='120' text-anchor='middle' x='90' y='155' font-family='sans-serif'>&#x1F464;</text>"
+        "</svg>"
+    )
+
+def _generate_initials(name):
+    if name:
+        names = name.split(" ")
+        initials = ''.join([x[0].upper() for x in names][::len(names) - 1 or 1])  # gets first and last or only first initial
+        return initials
+    return ''
```

Comment:
I don't like the initials approach. As already stated by @odony in the task, it might be impossible in some languages, it will be complex to handle all unicode fonts, and overall the purpose is not really reached for me. The goal is to make a person identifiable, and a color (which I think is fine) + a random image (which I suggest we use) offers a better visual clue than some letters. Letters in images are actually very distracting from main content.
See github, discord, for example ... (I think github does it better though, for discord there's no image variation only color).